signature/algorithm.go

```diff
@@ -0,0 +1,159 @@
+package signature
+
+import (
+	"errors"
+
+	"github.com/notaryproject/notation-core-go/signature"
+)
+
+// one of the following key spec name.
+//
+// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#algorithm-selection
+const (
+	RSA_2048 = "RSA-2048"
+	RSA_3072 = "RSA-3072"
+	RSA_4096 = "RSA-4096"
+	EC_256   = "EC-256"
+	EC_384   = "EC-384"
+	EC_521   = "EC-521"
+)
+
+// one of the following hash name.
+//
+// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#algorithm-selection
+const (
+	SHA_256 = "SHA-256"
+	SHA_384 = "SHA-384"
+	SHA_512 = "SHA-512"
+)
+
+// one of the following signing algorithm name.
+//
+// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#algorithm-selection
+const (
+	ECDSA_SHA_256      = "ECDSA-SHA-256"
+	ECDSA_SHA_384      = "ECDSA-SHA-384"
+	ECDSA_SHA_512      = "ECDSA-SHA-512"
+	RSASSA_PSS_SHA_256 = "RSASSA-PSS-SHA-256"
+	RSASSA_PSS_SHA_384 = "RSASSA-PSS-SHA-384"
+	RSASSA_PSS_SHA_512 = "RSASSA-PSS-SHA-512"
+)
+
+// InvalidKeySpec is the invalid value of keySpec.
+var InvalidKeySpec = signature.KeySpec{}
+
+// KeySpecName returns the name of a keySpec according to the spec.
+func KeySpecName(k signature.KeySpec) string {
+	switch k.Type {
+	case signature.KeyTypeEC:
+		switch k.Size {
+		case 256:
+			return EC_256
+		case 384:
+			return EC_384
+		case 521:
+			return EC_521
+		}
```

Comment:
should we fail other k.Size values here?

Reply:
> should we fail other k.Size values here?

Do you mean to return an error instead of an empty string?